ecommerce/frontend/src/components/Recommendations.js

```diff
@@ -0,0 +1,47 @@
+import './Recommendations.less'
+import { useEffect, useState } from 'react'
+import { formatProduct } from '../utils'
+import { notification, Spin } from 'antd'
+import { api } from '../api'
+import { ProductCard } from './product_card/ProductCard'
+import { useAppContext } from '../context/AppContext'
+
+export const Recommendations = () => {
+    const [isLoading, setIsLoading] = useState(true)
+    const [products, setProducts] = useState([])
+
+    useEffect(() => {
+        fetch()
+    }, [])
+
+    async function fetch() {
+        try {
+            setIsLoading(true)
+            const {
+                data: { status, products },
+            } = await api.get(`/recommendation`)
+            if (status.successful) {
+                setProducts(products.map(formatProduct))
+            } else {
+                notification.warning({ message: status.message })
+            }
+        } catch {
+            notification.error({ description: 'Failed to load recommendations' })
+        } finally {
+            setIsLoading(false)
+        }
+    }
+
+    return (
+        <Spin spinning={isLoading}>
+            <div className="recommendations-wrapper">
+                <h1 className="recommendations-header">Recommended Products For You</h1>
+                <div className="recommendations-content">
+                    {products.map(product => (
+                        <ProductCard key={product.id} product={product} />
+                    ))}
+                </div>
```

Comment:
In case the user is new and they don't have any recommendations, it would be nice to either hide this entire component or show something like "We currently do not have recommendations for your". I would prefer the first option.

Reply:
I discussed this with @furkannane, the response always has >0 # of products. Even the customer has no previous orders.